lib/cylc/task_events_mgr.py

```diff
@@ -1048,3 +1054,36 @@ def _reset_job_timers(self, itask):
         LOG.info('[%s] -%s', itask, message)
         # Set next poll time
         self.check_poll_time(itask)
+
+    @staticmethod
+    def process_execution_polling_delays(
+        delays, time_limit_delays, time_limit
+    ):
+        """Create list of intervals after starting at which to poll a task.
+
+        Args:
+            delays: Input is Execution Polling intervals.
+            time_limit_delays: Execution Time Limit Polling Intervals.
+            time_limit: Execution Time Limit.
+
+        Returns: List of delays from start of task.
+        """
+        # Remove excessive polling before time limit
+        while sum(delays) > time_limit:
+            del delays[-1]
+
+        # But fill up the gap before time limit
+        if delays:
```

Comment:
In the Cylc 8 PR, this logic was:

```python
        if sum(delays) > time_limit:
            # Remove execution polling which would overshoot the
            # execution time limit:
            while sum(delays) > time_limit:
                del delays[-1]
        elif delays:
```

What's the reason for the difference?

Reply:
Because @dpmatthews  added that logic in Cylc 8, but not Cylc 7 in this pr https://github.com/cylc/cylc-flow/pull/4844/files

I'm going to pull that logic in here.